src/crowd.go

```diff
@@ -1,37 +1,47 @@
 package main
 
 import (
+	"errors"
 	"flag"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"net/http"
 
-	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
-
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/14-bits/crowd/pkg/sink"
 	"github.com/julienschmidt/httprouter"
 )
 
 var port int
 var endpoint string
-var queue string
+var sinkType string
+var sinkUrl string
 
 func init() {
 	flag.IntVar(&port, "port", 2104, "HTTP Server Port")
 	flag.StringVar(&endpoint, "endpoint", "/api/foo", "HTTP path to receive POST requests")
-	flag.StringVar(&queue, "queue", "https://sqs.eu-central-1.amazonaws.com/21042018/foo", "URL for the SQS queue")
+	flag.StringVar(&sinkType, "sink-type", "void", "Sink adapter, e.g.: void, sqs, redis...")
+	flag.StringVar(&sinkUrl, "sink-url", "void", "Connection String/URL for the the sink")
 	flag.Parse()
 }
 
-type Queue struct {
-	Client sqsiface.SQSAPI
-	URL    string
+type Sink interface {
+	Push(payload []byte) error
 }
+
 type Crowd struct {
-	q Queue
+	s Sink
+}
+
+func SinkSelect(sinkType string, sinkUrl string) (Sink, error) {
+	switch sinkType {
+	case "sqs":
+		return sink.NewSQS(sinkUrl), nil
+	case "void":
+		return sink.NewVoid(sinkUrl), nil
+	default:
+		return nil, errors.New("Sink is not supported")
```

Comment:
Is it useful to return which sink is not supported in the error message?
ps.: did you receive a go linter warning for writing the first work in capital letters?

Reply:
I didn’t check, we can add the linter to the CI build to, I thought Capital was for public functions and lower for package unexported things ...

Reply:
It is ok! I don't really know what is behind `don't write your errors with capital letters`. 
From golang wiki: 

Error strings should not start with a capital letter because they'll often be prefixed before printing:

```go
err := TryInverseTachyonPulse()
if err != nil {
	fmt.Printf("failed to solve problem: %s\n", err)
}
```

BUT I really don't care about this one. I was just sharing this knowledge.

Reply:
Sorry, I understood it was the method name, not the error message. (huge jetlag here, sorry!)